src/pages/_app.js

```diff
@@ -1,10 +1,33 @@
-import GlobalStyle from "../../styles";
+import { useState } from "react";
+import { useRouter } from "next/router";
+import transactionsData from "../data/transactions.json";
 
 export default function App({ Component, pageProps }) {
+  const [transactions, setTransactions] = useState(transactionsData);
+  const [mode, setMode] = useState("default");
+  const [action, setAction] = useState("default");
+  const router = useRouter();
+
+  function handleCancel() {
```

Comment:
I think right now this is fine, but if you have more modes in the future you should consider using a handleMode function looking like this:

```js
function handleMode (mode){
setMode(mode)
}
```
This allows you to handle several modes in one function.

Reply:
thank for this goot hint! We will use it when we need it!

Reply:
implemented directly